manifests/init.pp

```diff
@@ -2,111 +2,112 @@
+#
+# @param manage_repos
+#   Trigger if files in /etc/yum.repos.d should get managed by Puppet exclusivly.
+#   If set to true, all unmanged files in /etc/yum.repos.d (and below) will get
+#   removed.
+#
+# @param repos_d_owner
+#   Set owner access to the /etc/yum.repos.d directory, representing a user.
+#
+# @param repos_d_group
+#   Set group access to the /etc/yum.repos.d directory, representing a group.
+#
+# @param repos_d_mode
+#   Set access permissions for the /etc/yum.repos.d directory, in numeric notation.
+#
+# @param repos_hiera_merge
+#   Trigger to merge all found instances of yum::repos in Hiera. This is useful
+#   for specifying repositories at different levels of the hierarchy and having
+#   them all included in the catalog.
+#
+# @param repos
+#   Hash of repos to pass to yum::repo. See yum::repo for more details.
+#
+# @param distroverpkg
+#   Trigger to add the distroverpkg setting to the main section of yum.conf.
+#   Will use the format distroverpk=$::operatingsystem-release (downcase) if active.
+#
+# @param pkgpolicy
+#   Trigger to add the pkgpolicy setting with the given string to the main section
+#   of yum.conf. Valid values are: 'newest' or 'last'.
+#
+# @param proxy
+#   Trigger to add the proxy setting with the given string to the main section of
+#   yum.conf. Specify a proxy URL.
+#
+# @param installonly_limit
+#   Trigger to add the installonly_limit setting with the given number to the main
+#   section of yum.conf. When unset (default) installonly_limit will not be present
+#   in yum.conf.
+#
+# @param exclude
+#   Value for the exclude setting in the main section of yum.conf.
+#   When undef (default) exclude will not be present in yum.conf.
+#
+# @param exclude_hiera_merge
+#   Trigger to merge all found instances of yum::exclude in Hiera. This is useful
+#   for specifying repositories at different levels of the hierarchy and having
+#   them all included in the catalog.
+#
 class yum (
-  $config_path         = '/etc/yum.conf',
-  $config_owner        = 'root',
-  $config_group        = 'root',
-  $config_mode         = '0644',
-  $manage_repos        = false,
-  $repos_d_owner       = 'root',
-  $repos_d_group       = 'root',
-  $repos_d_mode        = '0755',
-  $repos_hiera_merge   = true,
-  $repos               = undef,
-  $distroverpkg        = false,
-  $pkgpolicy           = undef,
-  $proxy               = undef,
-  $installonly_limit   = undef,
-  $exclude             = undef,
-  $exclude_hiera_merge = false,
+  Stdlib::Absolutepath $config_path           = '/etc/yum.conf',
+  String $config_owner                        = 'root',
+  String $config_group                        = 'root',
+  Pattern[/^[0-7]{4}$/] $config_mode          = '0644',
+  Boolean $manage_repos                       = false,
+  String $repos_d_owner                       = 'root',
+  String $repos_d_group                       = 'root',
+  Pattern[/^[0-7]{4}$/] $repos_d_mode         = '0755',
```

Comment:
Here's a better pattern

https://github.com/puppetlabs/puppetlabs-stdlib/commit/99bbe0ba5ea5c6141b3b4435bd2b0e7a77dd6c43#diff-b955865310c37e5eaa511c3abe074bdcR1

Hopefully that get's put in a tagged release soon and we can just use Stdlib::Filemode

Reply:
Have seen that before, should we wait until stdlib releases this ?

Reply:
Nah, let's not wait. We can use the pattern from there though.

Reply:
4.22.0 has `Stdlib::Filemode` now. Let's use that.

Reply:
changed